torch/_inductor/compile_fx.py

```diff
@@ -212,7 +212,10 @@ def get_static_input_idxs(num_fixed: int) -> list[int]:
     if not context or not context.fw_metadata:
         return fixed
 
-    return fixed + context.fw_metadata.static_input_indices
+    # fw_metadata contains the indices for the Dynamo graph. Since there are
+    # num_fixed inputs at the beginning of the AOT graph, we have to add
+    # num_fixed to get the positions correct.
+    return fixed + [num_fixed + i for i in context.fw_metadata.static_input_indices]
```

Comment:
Can we have aot autograd pass this more directly to inductor? This seems brittle / action at a distancey. Such as in the `meta` of the parameters.

Reply:
cc @bdhirsh

Reply:
> fw_metadata contains the indices for the Dynamo graph

I agree, and this seems wrong - AOTDispatcher should be ensuring that the indices here map directly to the inputs in the AOTDispatcher-generated forward graph